spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/FeignClientFactoryBean.java

```diff
@@ -120,9 +121,63 @@ public void setApplicationContext(ApplicationContext context) throws BeansExcept
 			builder.decode404();
 		}
 
+		// configure feign builder from properties if exists
```

Comment:
I think we could clean this code up a big.  Maybe have a method that extracts the properties whether from configuration beans or properties and then just sets them in the builder in one spot?